contracts/SpokePool.sol

```diff
@@ -1270,7 +1257,8 @@ abstract contract SpokePool is
 
         // slither-disable-next-line timestamp
         uint256 currentTime = getCurrentTime();
-        if (currentTime - quoteTimestamp > depositQuoteTimeBuffer) revert InvalidQuoteTimestamp();
```

Comment:
A future quote timestamp was causing underflow errors. I added this additional check to handle the issue. Should we introduce a separate error, `InvalidFutureQuoteTimestamp()`, or is it acceptable to combine both cases into `InvalidQuoteTimestamp()`?

Reply:
I think a future quote time causing under flow is intended. We don't want people using future quotes

Reply:
Yeah, that's right. Quote timestamps should not be in the future.

Reply:
I'm not sure why this has not come in (it might be due to merge issues) but this was erroring due to this PR not being in your branch:

https://github.com/across-protocol/contracts/pull/720/files

the solution done there was:

```
if (quoteTimestamp > currentTime || currentTime - quoteTimestamp > depositQuoteTimeBuffer)
            revert InvalidQuoteTimestamp();
```

the argument in that PR was that `underflow` OR `InvalidQuoteTimestamp` is bad error handling. After speaking with @pxrl it turns out this is often a problem for integrators and having a consistent error handling flow here is important.